### ISSUE TITLE
java10-dev not properly detected

### ISSUE THREAD

**Issue by nieder** [fink/fink#167]
On 10.13, I installed java10 (10.0.2) on a system that already had java16 and java9.

This is the output of `fink list system-java`:
````
$ fink list system-java
Information about 10164 packages read in 1 seconds.
 i   system-java                 10.0.2-1             [virtual package representing Java 10.0.2]
 i   system-java-dev             9.0.4-1              [virtual package representing Java SDK 10.0.2]
 i   system-java10               10.0.2-1             [virtual package representing Java 10]
     system-java10-dev           10.0.2-1             [virtual package representing Java 10 development headers]
 i   system-java14               1.4.2-1              [virtual package representing Java A]
 i   system-java14-dev           1.4.2-1              [virtual package representing Java A development headers]
 i   system-java15               1.5.0-1              [virtual package representing Java A]
 i   system-java15-dev           1.5.0-1              [virtual package representing Java A development headers]
 i   system-java16               1.6.0-1              [virtual package representing Java A]
 i   system-java16-dev           1.6.0-1              [virtual package representing Java A development headers]
     system-java17               1.7.0-1              [virtual package representing Java 1.7.0]
     system-java17-dev           1.7.0-1              [virtual package representing Java 1.7.0 development headers]
     system-java18               1.8.0-1              [virtual package representing Java 1.8.0]
     system-java18-dev           1.8.0-1              [virtual package representing Java 1.8.0 development headers]
     system-java3d               0-0                  [virtual package representing Java3D]
 i   system-java9                9.0.4-1              [virtual package representing Java 9]
 i   system-java9-dev            9.0.4-1              [virtual package representing Java 9 development headers]
     system-javaai               0-0                  [virtual package representing Java Advanced Imaging]
````

system-java10 is listed as present, but the -dev splitoff is not.

`fink-virtual-pkgs --debug` says this about Java:
````
- checking Java versions:
  - 10... jdk-10.0.2.jdk/Contents/Home/bin /Library/Java/JavaVirtualMachines/jdk-10.0.2.jdk/Contents/Home/10/Headers/jni.h missing 
  - 9... jdk-9.0.4.jdk/Contents/Home/bin jdk-9.0.4.jdk/Contents/Home/include 
  - A... A/Commands A/Headers/jni.h 
  - A... A/Commands A/Headers/jni.h 
  - A... A/Commands A/Headers/jni.h 
  - 1.7.0... not present on system
  - 1.8.0... not present on system
  - 9, x86_64 isn't recognized.  Contact the Fink Core Developers at fink-core@lists.sourceforge.net .
- searching for java3d... missing /System/Library/Java/Extensions/j3dcore.jar
- searching for javaai... missing /System/Library/Java/Extensions/jai_core.jar
````
jni.h is present in similar locations in both java9 and java10, so it should be found, but virtual-pkgs is looking in a non-existent `/10/` subdir.
````
$ find /Library/Java/JavaVirtualMachines/jdk-9.0.4.jdk/Contents/Home -name jni.h
/Library/Java/JavaVirtualMachines/jdk-9.0.4.jdk/Contents/Home/include/jni.h
$ find /Library/Java/JavaVirtualMachines/jdk-10.0.2.jdk/Contents/Home -name jni.h
/Library/Java/JavaVirtualMachines/jdk-10.0.2.jdk/Contents/Home/include/jni.h
````

**Comment by akhansen** [comment]
Because of the messy history and  of Java on OS X -> macOS, the detection code in VirtPackage.pm has to be manually adjusted constantly.  Java 10 isn't yet recognized as being similar to 9, but instead as something in the older versioned schemes.

**Comment by akhansen** [comment]
The version tests are in lines 427-435 of VirtPackage.pm .

**Comment by nieder** [comment]
I forgot to mention that I had tested adding `'10.0.Z, x86_64:	"Java SE 10"	/Library/Java/JavaVirtualMachines/jdk-10.0.Z.jdk/Contents/Home'`, at line 435 (after the 2nd java9 scheme) and it did not help detection.

**Comment by akhansen** [comment]
Addressed by #168.
